### DISCUSSION TITLE
What could be some reasons why the lock loop hangs?

### DISCUSSION THREAD

**Discussion by Kreijstal** [ccache/ccache#1412] [Support]
this loop https://github.com/ccache/ccache/blob/251f24409f80a711ca96ccd2288961cb4eb68aad/src/ccache/util/LockFile.cpp#L235
seems to hang forever on msys2, (not mingw), I guess it tries to lock something but can't, and it hangs forever? What could be some reasons it does this?
(this seems to hang the unittests)

**Comment by Kreijstal** [comment]
I see this happened before https://github.com/ccache/ccache/issues/170

**Comment by jrosdahl** [comment]
Sounds like a bug either in ccache or in MSYS2.

@Kreijstal: I suggest  [submitting a bug report](https://github.com/ccache/ccache/issues/new?assignees=&labels=bug&projects=&template=bug_report.md&title=) and then perhaps somebody interested in MSYS2 support can have a look at fixing it.

**Comment by Kreijstal** [comment]
solved with #1416
